File: examples/tcp_probe.erl
%% OpenClaw TCP port probe (Erlang) -- checks gateway nodes
-module(tcp_probe).
-export([main/0, probe/2]).

probe(Host, Port) ->
    case gen_tcp:connect(Host, Port, [binary, {active, false}], 3000) of
        {ok, Socket} ->
            gen_tcp:close(Socket),
            true;
        {error, _} ->
            false
    end.

main() ->
    Nodes = [{"localhost", 8080}, {"localhost", 8081}],
    lists:foreach(
        fun({Host, Port}) ->
            Status = case probe(Host, Port) of true -> "OK  "; false -> "FAIL" end,
            io:format("~s ~s:~p~n", [Status, Host, Port])
        end, Nodes).

File: examples/tcp_probe.exs
# OpenClaw TCP port probe (Elixir) — checks gateway nodes
defmodule TcpProbe do
  def probe(host, port, timeout \\ 3000) do
    opts = [:binary, active: false]

    case :gen_tcp.connect(String.to_charlist(host), port, opts, timeout) do
      {:ok, socket} ->
        :gen_tcp.close(socket)
        true

      {:error, _} ->
        false
    end
  end
end

nodes = [{"localhost", 8080}, {"localhost", 8081}]

Enum.each(nodes, fn {host, port} ->
  status = if TcpProbe.probe(host, port), do: "OK  ", else: "FAIL"
  IO.puts("#{status} #{host}:#{port}")
end)
